src/modules/sidebar/partials/ShareInternalLink.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div class="share-internal-link">
+    <div class="share-internal-link__icon">
+      <OpenInNew :size="20" />
+    </div>
+
+    <div class="share-internal-link__info">
+      <span class="share-internal-link__title">
+        {{ t('tables', 'Internal link') }}
+      </span>
+      <span class="share-internal-link__subtitle">
+        {{ t('tables', 'Only works for users with access to this folder') }}
+      </span>
+    </div>
+
+    <NcActionButton :title="copyTooltip" :aria-label="copyTooltip" @click="copyUrl" class="share-internal-link__button">
```

Comment:
NcActionButton is supposed to be only used within an NcActions component, probably worth to switch to https://nextcloud-vue-components.netlify.app/#/Components/NcButton which could save some custom styling